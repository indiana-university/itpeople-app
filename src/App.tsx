import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import Home from './components/Home';
import Page from './components/layout/Page';
import ProfileContainer from './components/ProfileContainer';
import Signin from './components/Signin';

const App : React.SFC = () => (
  <Page>
    <Switch>
      <Route path="/" exact={true} component={Home} />
      <Route path="/signin" component={Signin} />
      <Route path="/profile/:id" component={ProfileContainer} /> 
      <Route path="/me" component={ProfileContainer} /> 
    </Switch>
  </Page>
)

export default App;
