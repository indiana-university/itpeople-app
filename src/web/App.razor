@using RivetBlazor.Classes
@using RivetBlazor.Components
@inject Blazored.SessionStorage.ISessionStorageService sessionStorage
@inject NavigationManager Navigation

<Router AppAssembly="@typeof(Program).Assembly" PreferExactMatches="@true">
	<Found Context="routeData">
		<CascadingValue Value="@HeaderTitle">
			<CascadingValue Value="@NavLinks">
				<RouteView RouteData="@routeData" DefaultLayout="@typeof(RivetMainLayout)" />
			</CascadingValue>
		</CascadingValue>
	</Found>
	<NotFound>
		<CascadingValue Value="@HeaderTitle">
			<CascadingValue Value="@NavLinks">
				<LayoutView Layout="@typeof(RivetMainLayout)">
					<p>Sorry, there's nothing at this address.</p>
				</LayoutView>
			</CascadingValue>
		</CascadingValue>
	</NotFound>
</Router>
@code{
	public string HeaderTitle = "IT People";
	public List<IHeaderItem> NavLinks = new List<IHeaderItem>();
	private List<HeaderLink> ProfileLinks = new List<HeaderLink> { new HeaderLink("/AccountSettings", "Account Settings", "Manage your site preferences"), new HeaderLink("/AdminTool", "Admin Tool", "Manage other peoples' site preferences")};
	private AuthenticatedUser User = null;
	private string AuthenticationDestination;

	protected override async Task OnInitializedAsync()
	{
		User = await sessionStorage.GetItemAsync<AuthenticatedUser>("user");
		var loggedInUser = User?.Username ?? "";
		var isLoggedIn = User != null;
		
		AuthenticationDestination = new Uri(Navigation.Uri).PathAndQuery;
		var signInUrl = $"/SignIn?Destination={System.Net.WebUtility.UrlEncode(AuthenticationDestination)}";
		
		NavLinks = new List<IHeaderItem> {
			new HeaderLink("/People", "People"),
			new HeaderLink("/Units", "Units"),
			new HeaderProfile(loggedInUser, signInUrl, "/SignOut", ProfileLinks, null, isLoggedIn),
			//new HeaderProfile("bbatson", "/SignIn", "/SignOut", ProfileLinks, "Totally not Captain Marvel", true, avatar),
		};
	}
}