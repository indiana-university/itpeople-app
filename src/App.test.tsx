import { mount } from 'enzyme';
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import App from './App';

it('renders without crashing', () => {
  mount(
    <Router>
      <App />
    </Router>
  );
});

it('renders the home page by default', () => {
  const cut = mount(
    <Router>
      <App />
    </Router>
  );
  expect(cut.find("h1").text()).toBe("Home")
});
