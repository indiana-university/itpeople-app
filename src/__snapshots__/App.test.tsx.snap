// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the page title right 1`] = `
<h1
  className="rvt-ts-32 rvt-ts-41-lg-up rvt-text-bold"
>
  Test
</h1>
`;
